extensions/interactions/DragAndDropSortInput/directives/oppia-response-drag-and-drop-sort-input.component.ts

```diff
@@ -16,24 +16,39 @@
  * @fileoverview Component for the DragAndDropSortInput response.
  */
 
-require('services/html-escaper.service.ts');
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { DragAndDropAnswer } from 'interactions/answer-defs';
 
-angular.module('oppia').component('oppiaResponseDragAndDropSortInput', {
-  template: require('./drag-and-drop-sort-input-response.component.html'),
-  controllerAs: '$ctrl',
-  controller: ['$attrs', 'HtmlEscaperService',
-    function($attrs, HtmlEscaperService) {
-      var ctrl = this;
-      ctrl.chooseItemType = (index) => {
-        return (
-          index === 0 ?
-          'drag-and-drop-response-item' :
-          'drag-and-drop-response-subitem'
-        );
-      };
-      ctrl.$onInit = function() {
-        ctrl.answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
-        ctrl.isAnswerLengthGreaterThanZero = ctrl.answer.length > 0;
-      };
-    }]
-});
+@Component({
+  selector: 'oppia-response-drag-and-drop-sort-input',
+  templateUrl: './drag-and-drop-sort-input-response.component.html',
+  styleUrls: []
+})
+export class ResponseDragAndDropSortInputComponent implements OnInit {
+  @Input() answer!: string;
```

Comment:
Please add an explanatory comment why is the `!` added, you can find the comment in some component files that are already strictly typed.

Reply:
Done